addons/osfstorage/tests/test_views.py

```diff
@@ -649,6 +665,30 @@ def test_move_checkedout_file(self):
         )
         assert_equal(res.status_code, 405)
 
+    def test_within_node_move_while_preprint(self):
```

Comment:
Go ahead and make that. Might as well add an API test to disallow deleting a preprint file as well, since that's a case that also would have been caught if this had disappeared.